src/main/java/com/opencart/steps/SearchProductPageBL.java

```diff
@@ -25,26 +26,60 @@ public SearchProductPageBL makeSearch(String value) {
         searchProductPage.getInputSearchCriteria().clear();
         searchProductPage.getInputSearchCriteria().sendKeys(value);
         clickOnSearchGroupButton();
+        clickOnSelectButton();
+        clickOnSearchButton();
         searchProductResultPage = new SearchProductResultPage();
         return this;
     }
 
-    public void verifySearchResult(String searchCriteria) {
-        Assert.assertTrue(findAllProductBySearchCriteria(searchCriteria));
+    public void verifyPositiveSearchResult(String searchCriteria) {
+        SoftAssert softAssert = new SoftAssert();
+        String expectedResult = "Products meeting the search criteria";
+        String actualResult = searchProductResultPage.getSuccessTitle().getText();
+        softAssert.assertEquals(expectedResult, actualResult, "Title is correct, now you can check the list of products");
+        softAssert.assertTrue(findAllProductBySearchCriteria(searchCriteria),
+                "Check the list of products Done!");
+        softAssert.assertAll();
+    }
+
+    public void verifyNegativeSearchResult() {
+        SoftAssert softAssert = new SoftAssert();
+        String expectedResult = "There is no product that matches the search criteria.";
+        String actualResult = searchProductResultPage.getNoProductsFoundTitle().getText();
+        softAssert.assertEquals(expectedResult, actualResult, "Search criteria is actually correct");
```

Comment:
If you use only one assertion, please, use simple assertEquals, not SoftAssertion
Or, you can  add additional assertion here, to verify if the list of products is empty